### DISCUSSION TITLE
Is 1x turbo monorepo the right choice for my use case ?

### DISCUSSION THREAD

**Discussion by ricardo17coelho** [vercel/turborepo#5830] [Help]
Hi guys 🤚🏻

On my company we've several apps that belongs to the same "product".
Once we discovered Turbo, we had the idea to put all this app togther and share packages between this projects.

The project structure looks like this:

```
.
├── ...
├── apps                    # dir for the apps
│   ├── app-01              
│   ├── app-01-admin             
│   ├── app-02             
│   ├── app-02-admin 
│   ├── app-03            
│   └── ...             
├── packages                    # dir for the shared packages
│   ├── ui                      # ui packages built based on diff versions of the components, released by tag
        ├── accordion           # dir for ui component `accordion`, released inside of the `ui` and also as standalone package, released by tag
        ├── button              # same behaviour as `accordion`
        └── ...             
│   ├── something             
│   ├── something             
│   ├── something       
│   └── ...          
├── configs                    # dir for the shared configs
│   ├── eslint-config              
│   ├── release-it-config                   
│   └── ...   
└── ...
```

For the release of the packages, i set up `release-it`, which bumps the version of the packages, create an new commit and also creates an `git tag`.
Each project has his own pipeline which should be triggered depends on the needs.
For ex: `packages/ui` -> ci pipeline will be triggered when changes on the directory + tags ( regex that match the name of the tag)
( we release the packages/ui (and the components) to be possible to provide it to external apps (gitlab package registry)

At the moment i don't know how i can deploy the "apps", becase we should have diff `branches` (1 branch per ENV (like: `dev` or `int`)

Do someone have any ideia how the deployment process should work ?
Is turbo monorepo the right choice for this purpose or should we use multiple monorepos ?

**Comment by anthonyshew** [comment]
There are many ways to get your applications deployed. Turborepo wouldn't be a blocker here.

We have some [CI Recipes](https://turbo.build/repo/docs/ci) here but release strategies are different. The Vercel workflow will create previews for all pushes to your branches out-of-the-box but I'll leave this open for the community to talk about other strategies!
